server/chainquery/queries/claimQueries.js

```diff
@@ -0,0 +1,184 @@
+const logger = require('winston');
+
+const returnShortId = (claimsArray, longId) => {
+  let claimIndex;
+  let shortId = longId.substring(0, 1); // default short id is the first letter
+  let shortIdLength = 0;
+  // find the index of this claim id
+  claimIndex = claimsArray.findIndex(element => {
+    return element.claim_id === longId;
+  });
+  if (claimIndex < 0) {
+    throw new Error('claim id not found in claims list');
+  }
+  // get an array of all claims with lower height
+  let possibleMatches = claimsArray.slice(0, claimIndex);
+  // remove certificates with the same prefixes until none are left.
+  while (possibleMatches.length > 0) {
+    shortIdLength += 1;
+    shortId = longId.substring(0, shortIdLength);
+    possibleMatches = possibleMatches.filter(element => {
+      return (element.claim_id && (element.claim_id.substring(0, shortIdLength) === shortId));
+    });
+  }
+  return shortId;
+};
+
+const isLongClaimId = (claimId) => {
+  return (claimId && (claimId.length === 40));
+}
+
+const isShortClaimId = (claimId) => {
+  return (claimId && (claimId.length < 40));
+}
+
+export default (db, table) => ({
+
+  getShortClaimIdFromLongClaimId: async (claimId, claimName) => {
+    logger.debug(`claim.getShortClaimIdFromLongClaimId for ${claimName}#${claimId}`);
+    return await table.findAll({
+      where: { name: claimName },
+      order: [['height', 'ASC']],
+    }).then(result => {
+      if(result.length === 0) {
+        throw new Error('No claim(s) found with that claim name');
+      }
+
+      return returnShortId(result, claimId);
+    });
+  },
+
+  getAllChannelClaims: async (channelClaimId) => {
+    logger.debug(`claim.getAllChannelClaims for ${channelClaimId}`);
+    return await table.findAll({
+      where: { publisher_id: channelClaimId },
+      order: [['height', 'DESC']],
+      raw  : true,  // returns an array of only data, not an array of instances
+    })
+    .then(channelClaimsArray => {
+      if(channelClaimsArray.length === 0) {
+        return null;
+      }
+
+      return channelClaimsArray;
+    })
+  },
+
+  getClaimIdByLongChannelId: async (channelClaimId, claimName) => {
+    logger.debug(`finding claim id for claim ${claimName} from channel ${channelClaimId}`);
+    return await table.findAll({
+      where: { name: claimName, publisher_id: channelClaimId },
+      order: [['id', 'ASC']],
```

Comment:
any reason for ordering? `id` doesn't mean much so it may be wasteful to order the results, then again I think this is the default ordering for mysql anyway.

Reply:
Retained from the old query, I'll remove.